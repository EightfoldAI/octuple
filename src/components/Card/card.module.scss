.card {
  background-color: var(--card-background-color);
  border-radius: var(--card-border-radius);
  color: var(--card-text-color);
  display: block;
  font-family: var(--card-font-family);
  gap: $space-m;
  height: 260px;
  padding: $space-m;
  width: 360px;
  position: relative;

  &-bordered {
    border: 1px solid var(--card-border-color);
  }

  .icon {
    color: inherit;
  }

  .message {
    flex: 1;
    display: flex;
    align-items: center;
    color: inherit;
  }

  .buttons-container {
    padding: $space-m;
  }

  &-large {
    height: 278px;
    padding: $space-ml;
  }

  &-medium {
    height: 260px;
    padding: $space-m;
  }

  &-small {
    height: 260px;
    padding: $space-s;
  }

  &.disabled {
    opacity: $disabled-alpha-value;
    cursor: not-allowed;
  }

  &.drop-shadow {
    box-shadow: $shadow-object-m;
  }

  .header {
    @include octuple-h4();
    display: flex;
    flex-direction: row;
    margin-bottom: $space-ml;
    padding: $disabled-zero;

    .main-header {
      display: flex;
      .icon {
        height: fit-content;
        padding-top: $space-xs;
      }

      .title {
        margin-left: $space-xs;
      }

      &-button {
        margin-right: $space-l;
        padding: $disabled-zero;
        padding-bottom: $space-s;
        background-color: var(--card-background-color);
      }

      &-buttons {
        align-items: flex-end;
        align-self: start;
        height: fit-content;
        justify-content: right;
        white-space: nowrap;

        button {
          margin-left: $space-xs;
        }
      }
    }

    .sub-header {
      opacity: 50%;
      font-style: normal;
      font-weight: $text-font-weight-regular;
      font-size: $text-font-size-3;
      line-height: $text-line-height-3;
      display: flex;

      .separator {
        height: $space-xxs;
        width: $space-xxs;
        margin-right: $space-xs;
        margin-left: $space-xs;
        margin-top: $space-s;
      }
    }

    .button-icon {
      position: absolute;
      right: $disabled-zero;
      background-color: var(--card-background-color);
    }
  }

  .body {
    display: flex;
    flex-direction: column;
    overflow-y: auto;
    overflow-x: hidden;
    flex: 1;
    line-height: normal;

    .list {
      margin-bottom: $space-ml;

      &-item {
        display: flex;
      }

      ul {
        flex-wrap: wrap;
        display: -webkit-inline-box;
      }
    }
  }

  .footer {
    padding-top: $space-xs;
    bottom: $space-l;
    position: absolute;
    width: 100%;
    padding-right: 40px;

    .container {
      display: flex;
      justify-content: space-between;

      .content {
        display: flex;
      }
    }
  }
}

@import './Styles/rtl';
