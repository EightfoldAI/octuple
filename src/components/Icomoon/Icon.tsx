import React, { FC, useContext } from 'react';
import { mergeClasses } from '../../shared/utilities';
import { IcomoonProps, IcomoonSize } from './';
import IcomoonReact from 'icomoon-react';

import styles from './icon.module.scss';

const IconSetContext = React.createContext<object | null>(null);

export const IcomoonProvider = IconSetContext.Provider;

export const Icomoon: FC<IcomoonProps> = ({
    ariaHidden = false,
    classNames,
    color,
    iconName,
    id,
    role = 'presentation',
    size = IcomoonSize.Medium,
    'data-test-id': dataTestId,
}) => {
    const iconClassNames: string = mergeClasses([
        classNames,
        styles.iconWrapper,
    ]);

    return (
        <span
            data-test-id={dataTestId}
            aria-hidden={ariaHidden}
            className={iconClassNames}
            id={id}
            role={role}
        >
            <IconSetContext.Consumer>
                {(iconSet) => (
                    <IcomoonReact
                        iconSet={iconSet}
                        icon={iconName}
                        size={size}
                        color={color}
                    />
                )}
            </IconSetContext.Consumer>
        </span>
    );
};
