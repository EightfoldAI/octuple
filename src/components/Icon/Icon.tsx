'use client';

import React, { FC } from 'react';
import IcomoonReact from 'icomoon-react';

import { mergeClasses } from '../../shared/utilities';
import { Icon as MdiIcon } from '@mdi/react';
import { IconProps, IconSize } from './index';

import styles from './icon.module.scss';
import { useConfig } from '../ConfigProvider';

export const Icon: FC<IconProps> = ({
  ariaHidden = false,
  classNames,
  color,
  description,
  horizontal,
  id,
  path,
  role = 'presentation',
  rotate,
  size = IconSize.Medium,
  spin,
  style,
  title,
  vertical,
  'data-test-id': dataTestId,
  icomoonIconName,
}) => {
  const { icomoonIconSet } = useConfig();

  const iconClassNames: string = mergeClasses([classNames, styles.iconWrapper]);

  if (!icomoonIconName && !path) {
    return null;
  }

  const iconComponent = icomoonIconName ? (
    <IcomoonReact
      iconSet={icomoonIconSet}
      size={size}
      color={color}
      icon={icomoonIconName}
      role="presentation"
    />
  ) : (
    <MdiIcon
      color={color}
      description={description}
      horizontal={horizontal}
      path={path}
      rotate={rotate}
      size={size}
      title={title}
      vertical={vertical}
      spin={spin}
    />
  );

  return (
    <span
      data-test-id={dataTestId}
      aria-hidden={ariaHidden}
      className={iconClassNames}
      id={id}
      role={role}
      style={style ? style : null}
    >
      {iconComponent}
    </span>
  );
};

Icon.displayName = 'Icon';
