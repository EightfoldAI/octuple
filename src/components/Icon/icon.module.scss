.iconWrapper {
    display: inline-block;
    margin: 0;
    padding: 0;

    svg {
        vertical-align: middle;
    }
}