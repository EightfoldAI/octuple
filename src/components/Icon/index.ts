export * from './Icon';
export * from './Icon.types';
export * from './mdi';
