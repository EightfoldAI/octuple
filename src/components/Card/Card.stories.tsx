import React from 'react';
import { Stories } from '@storybook/addon-docs';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Card, CardSize } from './';
import { IconName } from '../Icon';
import { Avatar } from '../Avatar';
import { Pill } from '../Pills';
import { ButtonShape, SecondaryButton } from '../Button';

export default {
  title: 'Card',
  parameters: {
    docs: {
      page: (): JSX.Element => (
        <main>
          <article>
            <section>
              <h1>Card</h1>
              <p>
                A card can be used to display content. The content may consist
                of multiple elements in its header, body and footer.
              </p>
            </section>
            <section>
              <Stories includePrimary title="" />
            </section>
          </article>
        </main>
      ),
    },
  },
  argTypes: {
    size: {
      options: [CardSize.Flex, CardSize.Large, CardSize.Medium, CardSize.Small],
      control: { type: 'radio' },
    },
  },
} as ComponentMeta<typeof Card>;

const Card_Story: ComponentStory<typeof Card> = (args) => <Card {...args} />;

export const CustomCard = Card_Story.bind({});

// Storybook 6.5 using Webpack >= 5.76.0 automatically alphabetizes exports,
// this line ensures they are exported in the desired order.
// See https://www.npmjs.com/package/babel-plugin-named-exports-order
export const __namedExportsOrder = ['CustomCard'];

const baseCardArgs: Object = {
  dropShadow: true,
  size: CardSize.Medium,
  style: {},
  classNames: 'my-card-class',
};

CustomCard.args = {
  ...baseCardArgs,
  width: '360px',
  height: '520px',
  size: CardSize.Large,
  children: (
    <div style={{ textAlign: 'center', position: 'relative' }}>
      <Avatar
        theme="green"
        children="AB"
        type="round"
        size="80px"
        fontSize="36px"
        style={{ margin: '12px auto' }}
      />
      <div
        style={{
          marginBottom: '12px',
          fontSize: '24px',
          fontWeight: '600',
        }}
      >
        John Doe
      </div>
      <div
        style={{
          marginBottom: '34px',
          fontSize: '18px',
          fontWeight: '400',
        }}
      >
        Senior Quality Engineer
      </div>
      <div
        style={{
          display: 'flex',
          justifyContent: 'space-evenly',
          marginBottom: '24px',
          fontSize: '18px',
          fontWeight: '400',
        }}
      >
        <div>
          <div style={{ opacity: '50%', marginBottom: '5px' }}>
            Time in role
          </div>
          <div>2.5 y</div>
        </div>
        <div>
          <div style={{ opacity: '50%', marginBottom: '5px' }}>
            Skills | Skill gaps
          </div>
          <div>73 | 0</div>
        </div>
      </div>
      <div
        style={{
          textAlign: 'start',
          marginBottom: '34px',
          fontSize: '18px',
          fontWeight: '400',
          marginLeft: '35px',
        }}
      >
        <div
          style={{
            textAlign: 'start',
            marginBottom: '8px',
            fontSize: '18px',
            fontWeight: '400',
          }}
        >
          Last manager assessment
        </div>
        <div>Oct 1, 2022</div>
      </div>
      <div
        style={{
          display: 'flex',
          justifyContent: 'space-evenly',
          marginBottom: '48px',
        }}
      >
        <Pill label="Leadership Potential" theme="blueViolet" />
        <Pill label="High Risk" theme="violetRed" />
      </div>
      <div style={{ display: 'flex', justifyContent: 'space-evenly' }}>
        <SecondaryButton text="Add Development Plan" />
        <SecondaryButton
          iconProps={{ path: IconName.mdiDotsVertical }}
          shape={ButtonShape.Round}
        />
      </div>
    </div>
  ),
};
