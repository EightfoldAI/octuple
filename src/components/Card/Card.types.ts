import { IconName, IconProps } from '../Icon';
import { ButtonProps } from '../Button';
import { PillProps, PillType } from '../Pills';
import { OcBaseProps } from '../OcBase';
import { ConfigContextProps, Size } from '../ConfigProvider';
import { AvatarProps } from '../Avatar';

export enum CardSize {
  Flex = 'flex',
  Large = 'large',
  Medium = 'medium',
  Small = 'small',
}

export enum CardType {
  list = 'list',
  carousel = 'carousel',
}

export interface BodyListProps {
  showIcon?: boolean;
  label?: string;
}

export interface BodyProps {
  contents?: BodyListProps[];
  type?: string;
  iconProps?: IconProps;
}

export type FooterItems = [iconProps?: IconProps, text?: string];

export interface CardProps extends OcBaseProps<HTMLDivElement> {
  /**
   * Allows focus on the checkbox when it's disabled.
   */
  allowDisabledFocus?: boolean;
  /**
   * Avatar of the employee
   */
  avatar?: AvatarProps;
  /**
   * The Body of the dialog
   */
  body?: React.ReactNode;
  /**
   * Custom classes for the Body
   */
  bodyClassNames?: string;
  /**
   * Pill props for the first body list
   */
  bodyListOnePillProps?: PillProps;
  /**
   * Pill props for the second body list
   */
  bodyListTwoPillProps?: PillProps;
  /**
   * List of items for body
   */
  bodyListOneProps?: BodyProps;
  /**
   * List of items for body
   */
  bodyListTwoProps?: BodyProps;
  /**
   * If the card is bordered or not.
   * @default false
   */
  bordered?: boolean;
  /**
   * Custom classnames of the component.
   */
  classNames?: string;
  /**
   * Configure how contextual props are consumed
   */
  configContextProps?: ConfigContextProps;
  /**
   * The card is in a disabled state.
   * @default false
   */
  disabled?: boolean;
  /**
   * The card has a drop shadow
   * @default false
   */
  dropShadow?: boolean;
  /**
   * The Footer of the dialog
   */
  footer?: React.ReactNode;
  /**
   * Props for the Footer button
   */
  footerButtonProps?: ButtonProps;
  /**
   * Custom classes for the Footer
   */
  footerClassNames?: string;
  /**
   * Footer icon name
   */
  footerIcon?: IconName;
  /**
   * The props for the footer
   */
  footerProps?: FooterItems;
  /**
   * The Header of the dialog
   */
  header?: React.ReactNode;
  /**
   * Props for the header button
   */
  headerButtonProps?: ButtonProps;
  /**
   * Custom classes for the header
   */
  headerClassNames?: string;
  /**
   * Header icon name
   */
  headerIcon?: IconName;
  /**
   * Header title name
   */
  headerTitle?: string;
  /**
   * Custom height of the panel
   */
  height?: number;
  /**
   * Custom icon for the card
   * @default IconName.mdiInformation | IconName.mdiCheckCircle | IconName.mdiAlert
   */
  icon?: IconName;
  /**
   * The boolean value if a list card is selected
   */
  isSelected?: boolean;
  /**
   * Name of the employee card
   */
  name?: string;
  /**
   * The checkbox size.
   * @default SelectorSize.Medium
   */
  size?: CardSize | Size;
  /**
   * The card style.
   */
  style?: React.CSSProperties;
  /**
   * The props for the subheaders
   */
  subHeaderProps?: Array<string>;
  /**
   * Custom icon subheader separator
   */
  subHeaderSeparatorIcon?: IconName;
  /**
   * Tags for the employee
   */
  tag?: PillProps;
  /**
   * Type of the card
   * @default CardType.neutral
   */
  type?: CardType;
  /**
   * Custom width of the panel
   */
  width?: number;
}
