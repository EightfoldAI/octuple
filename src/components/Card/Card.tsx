'use client';

import React, { FC, Ref, useContext } from 'react';
import DisabledContext, { Disabled } from '../ConfigProvider/DisabledContext';
import { CardProps, CardSize, CardType } from './Card.types';
import { mergeClasses } from '../../shared/utilities';
import { ButtonShape, TwoStateButton } from '../Button';
import { SizeContext, Size } from '../ConfigProvider';
import { Icon } from '../Icon';
import { Stack } from '../Stack';
import { Pill } from '../Pills';
import { List } from '../List';
import { Breakpoints, useMatchMedia } from '../../hooks/useMatchMedia';
import { useCanvasDirection } from '../../hooks/useCanvasDirection';

import styles from './card.module.scss';

export const Card: FC<CardProps> = React.forwardRef(
  (
    {
      allowDisabledFocus = false,
      avatar,
      body,
      bodyClassNames,
      bodyListOnePillProps,
      bodyListTwoPillProps,
      bodyListOneProps,
      bodyListTwoProps,
      bordered = false,
      children,
      classNames,
      configContextProps = {
        noDisabledContext: false,
        noSizeContext: false,
      },
      disabled = false,
      dropShadow = false,
      footer,
      footerClassNames,
      footerProps,
      header,
      headerButtonProps,
      headerClassNames,
      headerIcon,
      headerTitle,
      height,
      icon,
      insetFocusVisible = false,
      isSelected = false,
      name,
      size = CardSize.Medium,
      style,
      subHeaderProps,
      subHeaderSeparatorIcon,
      type = CardType.list,
      width,
      ...rest
    },
    ref: Ref<HTMLDivElement>
  ) => {
    const largeScreenActive: boolean = useMatchMedia(Breakpoints.Large);
    const mediumScreenActive: boolean = useMatchMedia(Breakpoints.Medium);
    const smallScreenActive: boolean = useMatchMedia(Breakpoints.Small);
    const xSmallScreenActive: boolean = useMatchMedia(Breakpoints.XSmall);

    const htmlDir: string = useCanvasDirection();

    const contextuallySized: Size = useContext(SizeContext);
    const mergedSize = configContextProps.noSizeContext
      ? size
      : contextuallySized || size;

    const contextuallyDisabled: Disabled = useContext(DisabledContext);
    const mergedDisabled: boolean = configContextProps.noDisabledContext
      ? disabled
      : contextuallyDisabled || disabled;

    const cardClasses: string = mergeClasses([
      classNames,
      styles.card,
      {
        [styles.cardBordered]: !!bordered,
      },
      {
        [styles.cardSmall]: mergedSize === CardSize.Flex && largeScreenActive,
      },
      {
        [styles.cardMedium]: mergedSize === CardSize.Flex && mediumScreenActive,
      },
      {
        [styles.cardMedium]: mergedSize === CardSize.Flex && smallScreenActive,
      },
      {
        [styles.cardLarge]: mergedSize === CardSize.Flex && xSmallScreenActive,
      },
      { [styles.list]: type === CardType.list },
      { [styles.disabled]: allowDisabledFocus || mergedDisabled },
      { [styles.cardLarge]: mergedSize === CardSize.Large },
      { [styles.cardMedium]: mergedSize === CardSize.Medium },
      { [styles.cardSmall]: mergedSize === CardSize.Small },
      { [styles.dropShadow]: dropShadow },
      { [styles.insetFocusVisible]: insetFocusVisible },
      { [styles.cardRtl]: htmlDir === 'rtl' },
    ]);

    const headerClasses: string = mergeClasses([
      styles.header,
      headerClassNames,
    ]);

    const bodyClasses: string = mergeClasses([styles.body, bodyClassNames]);

    const footerClasses: string = mergeClasses([
      styles.footer,
      footerClassNames,
    ]);

    return (
      <div
        {...rest}
        className={cardClasses}
        ref={ref}
        style={{ ...style, ...{ height, width } }}
      >
        {children ? (
          children
        ) : (
          <>
            {/** TODO: Update predefined variants to meet latest specification. */}
            <div className={headerClasses}>
              {header && header}
              {(icon || headerTitle) && (
                <div className={styles.mainHeader}>
                  {icon && (
                    <Icon path={icon} classNames={styles.icon} size={'40px'} />
                  )}
                  {headerTitle && (
                    <div className={styles.title}>
                      {headerTitle}
                      <div className={styles.subHeader}>
                        {subHeaderProps &&
                          subHeaderProps.map((item, idx) => {
                            return (
                              <>
                                <div>{item}</div>
                                {idx < subHeaderProps.length - 1 && (
                                  <Icon
                                    path={subHeaderSeparatorIcon}
                                    classNames={styles.separator}
                                  />
                                )}
                              </>
                            );
                          })}
                      </div>
                    </div>
                  )}
                </div>
              )}
              {headerButtonProps && (
                <div className={styles.buttonIcon}>
                  <TwoStateButton
                    classNames={styles.mainHeaderButton}
                    shape={ButtonShape.Round}
                    iconOneProps={headerButtonProps.iconProps}
                    {...headerButtonProps}
                  />
                </div>
              )}
            </div>
            <div className={bodyClasses}>
              {bodyListOneProps && (
                <List
                  layout="horizontal"
                  classNames={styles.list}
                  itemStyle={{ margin: '5px' }}
                  items={bodyListOneProps.contents}
                  renderItem={(item) => {
                    return bodyListOneProps.type === 'list' ? (
                      <div className={styles.listItem}>
                        {item.showIcon && (
                          <Icon {...bodyListOneProps.iconProps} />
                        )}
                        <span>{item.label}</span>
                      </div>
                    ) : (
                      <Pill label={item.label} {...bodyListOnePillProps} />
                    );
                  }}
                />
              )}
              {bodyListTwoProps && (
                <Stack direction="horizontal" gap="xs" wrap="wrap">
                  {bodyListTwoProps?.contents.map((item) => {
                    return bodyListTwoProps.type === 'list' ? (
                      <div className={styles.listItem}>
                        {item.showIcon && (
                          <Icon {...bodyListOneProps.iconProps} />
                        )}
                        <span>{item.label}</span>
                      </div>
                    ) : (
                      <Pill label={item.label} {...bodyListTwoPillProps} />
                    );
                  })}
                </Stack>
              )}
              {body && body}
            </div>
            <div className={footerClasses}>
              {footerProps && (
                <div className={styles.container}>
                  {footerProps.map((item: any) => {
                    return (
                      <div className={styles.content}>
                        <Icon {...item.iconProps} />
                        <div>{item.text}</div>
                      </div>
                    );
                  })}
                </div>
              )}
              {footer && footer}
            </div>
          </>
        )}
      </div>
    );
  }
);
