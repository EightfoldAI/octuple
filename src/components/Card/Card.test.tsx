import React from 'react';
import MatchMediaMock from 'jest-matchmedia-mock';
import Enzyme, { mount, ReactWrapper } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Card } from './Card';
import { IconName } from '../Icon';
import { render } from '@testing-library/react';
import { CardSize } from './Card.types';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

describe('Card', () => {
  let wrapper: ReactWrapper;
  const footer = 'This is the card footer';
  const body = 'This is the card body';
  const header = 'This is the card header';

  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });
  afterEach(() => {
    matchMedia.clear();
  });

  beforeEach(() => {
    wrapper = mount(<Card />);
  });

  test('card content', () => {
    wrapper.setProps({
      footer,
      header,
      body,
    });
    expect(wrapper.find('.footer').text()).toBe(footer);
    expect(wrapper.find('.body').text()).toBe(body);
    expect(wrapper.find('.header').text()).toBe(header);
  });

  test('card header actions exist', () => {
    wrapper.setProps({
      footer,
      header,
      body,
      headerButtonProps: {
        classNames: 'header-button',
        iconProps: { path: IconName.mdiArrowLeftThick },
      },
    });
    expect(wrapper.find('.header-button').length).toBeTruthy();
  });

  test('Card is large', () => {
    const { container } = render(
      <Card body={body} footer={footer} header={header} size={CardSize.Large} />
    );
    expect(container.getElementsByClassName('card-large')).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });

  test('Card is medium', () => {
    const { container } = render(
      <Card
        body={body}
        footer={footer}
        header={header}
        size={CardSize.Medium}
      />
    );
    expect(container.getElementsByClassName('card-medium')).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });

  test('Card is small', () => {
    const { container } = render(
      <Card body={body} footer={footer} header={header} size={CardSize.Small} />
    );
    expect(container.getElementsByClassName('card-small')).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });
});
