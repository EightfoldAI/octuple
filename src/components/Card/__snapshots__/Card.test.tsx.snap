// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card is large 1`] = `
<div>
  <div
    class="card list card-large"
  >
    <div
      class="header"
    >
      This is the card header
    </div>
    <div
      class="body"
    >
      This is the card body
    </div>
    <div
      class="footer"
    >
      This is the card footer
    </div>
  </div>
</div>
`;

exports[`Card Card is medium 1`] = `
<div>
  <div
    class="card list card-medium"
  >
    <div
      class="header"
    >
      This is the card header
    </div>
    <div
      class="body"
    >
      This is the card body
    </div>
    <div
      class="footer"
    >
      This is the card footer
    </div>
  </div>
</div>
`;

exports[`Card Card is small 1`] = `
<div>
  <div
    class="card list card-small"
  >
    <div
      class="header"
    >
      This is the card header
    </div>
    <div
      class="body"
    >
      This is the card body
    </div>
    <div
      class="footer"
    >
      This is the card footer
    </div>
  </div>
</div>
`;
